cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1901,14 +2158,29 @@ class device_row_hasher {
     {
     }
 
-    template <typename T, CUDF_ENABLE_IF(not cudf::is_nested<T>())>
+    /**
+     * @brief Dummy operator for dispatch to void type. Ideally, we want this to be unreachable, but
+     * using CUDF_UNREACHABLE leads to an increase in register usage and is avoided.
```

Comment:
question: Did local memory or stack memory usage change when using `CUDF_UNREACHABLE`? Telling the compiler that this code path is unreachable may have allowed it to better inline which may have increased register usage, but may have also decreased stack/lmem usage.

Reply:
Good to know! Will investigate this

Reply:
I tested this out by looking at the ncu profiles for case when I use `CUDF_UNREACHABLE` vs when I return a dummy value. I see that the number of requests to lmem is 0 is both cases